test/jdk/java/awt/Dialog/DialogLocationTest.java

```diff
@@ -0,0 +1,176 @@
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+  @test
+  @bug 4101437
+  @summary Dialog.setLocation(int,int) works unstable when the dialog is visible
+  @key headful
+  @run main DialogLocationTest
+*/
+
+import java.awt.AWTException;
+import java.awt.Container;
+import java.awt.Dialog;
+import java.awt.EventQueue;
+import java.awt.Frame;
+import java.awt.GraphicsEnvironment;
+import java.awt.Panel;
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.awt.Robot;
+import java.awt.event.ComponentAdapter;
+import java.awt.event.ComponentEvent;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Random;
+
+public class DialogLocationTest extends Panel {
+    private volatile int count = 0;
+    private Dialog my_dialog;
+    private volatile boolean waitingForEvent = false;
+    private volatile int newX, newY;
+    Random random = new Random();
+
+    public void init() {
+        Container f = getParent();
+
+        while (!(f instanceof Frame)) {
+            f = f.getParent();
+        }
+
+        my_dialog = new Dialog((Frame) f, "TestDialog");
+        my_dialog.setSize(150, 100);
+
+        setSize(200, 200);
+    }
+
+    public void start() throws InterruptedException,
+            InvocationTargetException {
+        Robot robot;
+        try {
+            robot = new Robot();
+            EventQueue.invokeAndWait(() -> {
+                my_dialog.setLocationRelativeTo(null);
+                my_dialog.setVisible(true);
+            });
+            robot.waitForIdle();
+            robot.delay(1000);
+            my_dialog.addComponentListener(new CL());
+            setDialogLocation(my_dialog);
+        } catch (AWTException e) {
+            throw new RuntimeException(e);
+        } finally {
+            EventQueue.invokeAndWait(() -> {
+                my_dialog.setVisible(false);
+                my_dialog.dispose();
+            });
+        }
+    }
+
+    public void setDialogLocation(Dialog dialog) {
+        int height, width, insetX, insetY;
+        Point curLoc;
+        int i;
+
+        Rectangle screen = GraphicsEnvironment
+                .getLocalGraphicsEnvironment()
+                .getMaximumWindowBounds();
+        height = screen.height;
+        width = screen.width;
+        insetX = screen.x;
+        insetY = screen.y;
+
+        for (i = 0; i < 100; i++) {
+            newX = random.nextInt(width - 300) +  insetX;
+            newY = random.nextInt(height - 400) + insetY;
```

Comment:
Your version will get a number between insetX and insetX+width-300.
The original change: between insetX and width-300.

Reply:
fixed to
newX = random.nextInt(width - 300 - insetX) +  insetX;
newY = random.nextInt(height - 400 - insetY) + insetY;